### ISSUE TITLE
runtime/pprof: TestTraceStress crashes with SIGBUS on freebsd-386-gce101

### ISSUE THREAD

**Issue by mikioh** [golang/go#9901]
$ go version
go version devel +82ee85e w/ change 4330

$ environment
trybot: freebsd-386-gce101, which runs freebsd/386 runtime on freebsd/amd64 kernel.

See See https://storage.googleapis.com/go-build-log/5c323c50/freebsd-386-gce101_fca8db55.log for further information.

The headline of dying message (SIGBUS) is as follows:
```
fatal error: unexpected signal during runtime execution
[signal 0xa code=0xc addr=0x28306000 pc=0x805ac86]
```
Not sure this is equivalent to #9729.

**Comment by dvyukov** [comment]
@RLH @aclements This seems to be more related to write barriers rather than tracing. TestTraceStress just stresses runtime well.

**Comment by mikioh** [comment]
Merged into #9953
